src/util/measurementUncertaintyTool.js

```diff
@@ -11,56 +11,50 @@ import Decimal from 'decimal.js';
-      Uncertainty rounding measurement rule
-      regarding first significant figure:
-      if the first significant figure is greater than 2,
-      rounding digit stays in the same index.
-      However if it is 1 or 2, add one more 0 in the decimal.
-      **/
-
-    for (i = 0; i <= belowOne.length; i++) {
-      if (belowOne[i] === '0') {
-        continue;
-      } else if (belowOne[i] === '1' || belowOne[i] === '2') {
-        const powerOf = new Decimal(-(i + 2)).abs();
-        const decimalPlace = parseInt(powerOf);
-        const roundedValue = value.toFixed(decimalPlace);
+function getPixelDiagonal(colPixelSpacing, rowPixelSpacing) {
+  if (colPixelSpacing !== undefined && colPixelSpacing !== null) {
+    return new Decimal(colPixelSpacing ** 2 + rowPixelSpacing ** 2).sqrt();
+  }
 
-        return [roundedValue, powerOf]; // PowerOf is used to fix the digit
-      } else {
-        const powerOf = new Decimal(-(i + 1)).abs();
-        const decimalPlace = parseInt(powerOf);
-        const roundedValue = value.toFixed(decimalPlace);
+  return Decimal.sqrt(2);
+}
 
-        return [roundedValue, powerOf];
-      }
-    }
-  } else {
-    // When the value is larger than 1
-    const aboveOne = uncertaintyValue.toString().split('.')[0];
+function roundValueBasedOnUncertainty(inputValue, uncertaintyValue) {
+  const valueToBeRounded = new Decimal(inputValue);
+  const roundingRange = getRoundingRange(uncertaintyValue);
 
-    if (aboveOne[0] === '1' || aboveOne[0] === '2') {
-      const powerOf = aboveOne.length - 2;
-      const roundingRange = Decimal.pow(10, powerOf);
-      const roundedValue = new Decimal(value).toNearest(roundingRange);
+  return uncertaintyValue < 1
+    ? valueToBeRounded.toFixed(roundingRange)
+    : valueToBeRounded.toNearest(roundingRange);
+}
 
-      return [roundedValue, Decimal.abs(powerOf)];
-    }
-    const powerOf = aboveOne.length - 1;
-    const roundingRange = Decimal.pow(10, powerOf);
-    const roundedValue = new Decimal(value).toNearest(roundingRange);
+function getThefirstSFIndex(uncertaintyValue) {
+  return Decimal.ceil(-Decimal.log10(uncertaintyValue));
+}
 
-    return [roundedValue, Decimal.abs(powerOf)];
+function getRoundingRange(uncertaintyValue) {
+  if (uncertaintyValue < 1) {
+    const firstSFIndex = getThefirstSFIndex(uncertaintyValue);
+    const firstSignificantDigitValue = uncertaintyValue
+      .toString()
+      .charAt(firstSFIndex.plus(1));
+
+    return firstSignificantDigitValue < 3
+      ? parseInt(firstSFIndex.plus(1), 10)
+      : parseInt(firstSFIndex, 10);
   }
+  const uncertaintyGreaterThanOne = uncertaintyValue.toString().split('.')[0];
+  const powerOf =
+    uncertaintyGreaterThanOne[0] < 3
+      ? uncertaintyGreaterThanOne.length - 2
+      : uncertaintyGreaterThanOne.length - 1;
+
+  return Decimal.pow(10, powerOf);
 }
 
-export { roundValue };
+export {
+  roundValueBasedOnUncertainty,
+  getPixelDiagonal,
+  getThefirstSFIndex,
+  getRoundingRange,
```

Comment:
do the last two functions need to be exported?

Reply:
for the unit test purpose